modules/azure-vmcluster/main.tf

```diff
@@ -0,0 +1,69 @@
+terraform {
+  required_providers {
+    azapi = {
+      source = "Azure/azapi"
+    }
+  }
+}
+
+
+data "azapi_resource_list" "listDbServersByCloudExadataInfrastructure" {
+  type                   = "Oracle.Database/cloudExadataInfrastructures/dbServers@2023-09-01"
+  parent_id              = var.exadata_infrastructure_id
+  response_export_values = ["*"]
+}
+
+
+locals {
+  exa_infra_dbServers       = jsondecode(data.azapi_resource_list.listDbServersByCloudExadataInfrastructure.output).value[*]
+  exa_infra_dbServers_ocids = toset([for dbs in local.exa_infra_dbServers : "${dbs.properties.ocid}"])
+
+
+  # use supplied dbservers if provided else use exa.dbservers
+  are_dbservers_provided = (length(var.exadata_infra_dbserver_ocids) == 0)
+  db_server_ocids        = local.are_dbservers_provided ? local.exa_infra_dbServers_ocids : var.exadata_infra_dbserver_ocids
```

Comment:
The logic looks good. `are_dbservers_provided = (length == 0)` in line 23 is a little bit confusing though.

Reply:
done!